ScreenCapturerExample/ExampleWebViewSource.swift

```diff
@@ -93,38 +94,101 @@ class ExampleWebViewSource: NSObject {
+                                          orientation: TVIVideoOrientation.up,
+                                          timestamp: timer.timestamp)
+            } else if let theError = error {
+                print("Snapshot error:", theError as Any)
+            }
+        })
+    }
+
+    private func deliverCapturedImage(image: UIImage,
+                                      orientation: TVIVideoOrientation,
+                                      timestamp: CFTimeInterval) {
+        /*
+         * Make a (deep) copy of the UIImage's underlying data. We do this by getting the CGImage, and its CGDataProvider.
+         * In some cases, the bitmap's pixel format is not compatible with CVPixelBuffer and we need to repack the pixels.
+         */
+        guard let cgImage = image.cgImage else {
+            return
         }
 
+        let alphaInfo = cgImage.alphaInfo
+        let byteOrderInfo = CGBitmapInfo(rawValue: cgImage.bitmapInfo.rawValue & CGBitmapInfo.byteOrderMask.rawValue)
+        let dataProvider = cgImage.dataProvider
+        let data = dataProvider?.data
+        let baseAddress = CFDataGetBytePtr(data!)!
         /*
-         * Make a copy of the UIImage's underlying data. We do this by getting the CGImage, and its CGDataProvider.
-         * Note that this technique is inefficient because it causes an extra malloc / copy to occur for every frame.
-         * For a more performant solution, provide a pool of buffers and use them to back a CGBitmapContext.
+         * The underlying data is marked as immutable, but we are the sole owner and can do as we please.
+         * Also, the CVPixelBuffer constructor will only accept a mutable pointer.
          */
-        let image: CGImage? = contextImage?.cgImage
-        let dataProvider: CGDataProvider? = image?.dataProvider
-        let data: CFData? = dataProvider?.data
-        let baseAddress = CFDataGetBytePtr(data!)
-        contextImage = nil
+        let mutableBaseAddress = UnsafeMutablePointer<UInt8>(mutating: baseAddress)
+        var pixelFormat = TVIPixelFormat.format32BGRA
+
+        var imageBuffer = vImage_Buffer(data: mutableBaseAddress,
+                                        height: vImagePixelCount(cgImage.height),
+                                        width: vImagePixelCount(cgImage.width),
+                                        rowBytes: cgImage.bytesPerRow)
+
+        switch byteOrderInfo {
+        case .byteOrder32Little:
+            // Pixel format encountered on iOS simulators.
+            // Note: We have observed that pre-multiplied images might contain non-opaque alpha on the simulator.
+            assert(alphaInfo == .premultipliedFirst || alphaInfo == .noneSkipFirst)
+            if (alphaInfo == .premultipliedFirst) {
+                // TODO: This vImage call is crashing.
+//                vImageUnpremultiplyData_ARGB8888(&imageBuffer, &imageBuffer, vImage_Flags(kvImageDoNotTile))
```

Comment:
formatting is off. 
Whats our strategy here?

Reply:
For comparison, here is the app running on my iPhone XS.

![img_1715](https://user-images.githubusercontent.com/1302577/51066664-655cde80-15c0-11e9-8d4d-954c9a5c8e53.PNG)